### PR TITLE
[21561] Bump version to 3.0.1

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR bumps `Shapes Demo docs` version to `v3.0.1` and updates the notes.

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- **N/A** Documentation tests pass locally.
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] CI passes without warnings or errors.

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/shapes_demo_docs_manual/224/)

File: docs/notes/notes.rst [hunk 1]
```diff
@@ -1,34 +1,16 @@
...
-#. Move DR ``TypeConsistencyEnforcement`` & ``DataRepresentation`` from ``TypeConsistency`` to ``DataReaderQos``.
-#. Add CDRMessage_t include.
-#. Use token for CCache action.
-#. Migrate fastrtps namespace to fastdds.
-#. Adjust for ``const`` qualification of all data related inputs in DataWriter APIs.
-#. Adjust for Fast DDS public headers migration to .hpp.
-#. Improve Github CI.
-#. Remove new line character at the end of ``SHAPESVERSION`` CMake variable
-#. Regenerate types with Fast DDS-Gen v4
+#. Support to Fast DDS 3.0.1.
```
Comment (JesusPoderoso) [inline docs/notes/notes.rst:6]:
```suggestion
#. Support to Fast DDS v3.0.1.
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
NIT

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/shapes_demo_docs_manual/225/)

**Pull request merged**